constraints.txt

```diff
@@ -0,0 +1,135 @@
+absl-py==1.4.0
```

Comment:
Hey -- we need to have a bit more of a holistic conversation here, probably a document describing the user journeys would be best. I ABSOLUTELY think pinning things absolutely has a place! But we need to figure out the place.

Zooming out, we've offered two modes of maxtext historically, MaxText on JaxNightly and JaxStable. (And we've tested both etc.)

I think @chajath's observation is we woudl like to offer a third mode, MaxText on Pinned. For MaxText on pinned, we would have everything pinned. And there would need to be a periodic process of turning a "golden" JaxStable into a Pinned version.

(Separately, we might also experiment with having MaxText releases that freeze a MaxText commit. But either way a MaxText commit + MODE=pinned == rock solid point in time).

But I'm very hesitant to make changes without a bit of a clearer plan. (Who updrades the pinned closure? When? What is the playbook?) And fundamental engineering support (it seems to me like pinned is an alternative to stable/nightly so it should be a supported MODE?).

And I've been getting pings on this from @chajath and Aman Gupta. I'll intro you two via google chat.

Reply:
Following up on the offline discussion - let's punt renaming build modes as a follow-up task, and for now let's settle on `nightly`, `stable`, and `pinned` modes. Could you take another look, please? Thanks!